lib/charms/opensearch/v0/opensearch_users.py

```diff
@@ -57,27 +63,40 @@ def create_role(
             raise OpenSearchUserMgmtError(f"creating role {role_name} failed - response: {resp}")
         return resp
 
-    def remove_role(self, role_name: str) -> None:
+    def remove_role(self, role_name: str) -> Dict[str, any]:
         """Remove the given role from opensearch distribution.
 
         Args:
             role_name: name of the role to be removed.
 
         Raises:
             OpenSearchUserMgmtError: If the request fails, or if role_name is empty
+
+        Returns:
+            HTTP response to opensearch API request.
         """
         if not role_name:
             raise OpenSearchUserMgmtError(
                 "role name empty - sending a DELETE request to endpoint root isn't permitted"
             )
 
-        resp = self.opensearch.request("DELETE", f"{ROLE_ENDPOINT}/{role_name}")
+        try:
+            resp = self.opensearch.request("DELETE", f"{ROLE_ENDPOINT}/{role_name}")
+        except OpenSearchHttpError as e:
+            if e.status_code == 404:
+                return {
```

Comment:
Why is the returned object needed? for logging?

Reply:
It's really for consistency with the rest of the library - If we're sending requests I feel we should return the responses. In this case, returning this fake response allows a caller function to be aware that the role never existed in the first place, so no action is being taken.

Reply:
So far, this is only for internal use and has nothing to do with the client (requirer) app. Its intended use is to remove roles when the relation is broken, or when we're removing lingering users. If we're trying to remove users that don't exist, I want to return a positive signal that the user doesn't exist, so the caller function can continue with the knowledge that user definitely doesn't exist. However, I forgot to add `else: raise` to this function, so I've added that in